### ISSUE TITLE
Version 5.0.0 asks for ansible module even when it is installed

### ISSUE THREAD

**Issue by balmasea** [ansible-community/ansible-lint#1330]
##### Summary
Latest version of ansible_lint is crashing our pipelines.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console (paste below)
$ ansible --version
ansible 2.9.9
  config file = None
  configured module search path = ['/home/{user}/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/{user}/GNAT/ansible-pb-text_based_backups/venv/lib/python3.6/site-packages/ansible
  executable location = /home/{user}/GNAT/ansible-pb-text_based_backups/venv/bin/ansible
  python version = 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
  
  
ansible-lint --version
ansible-lint 5.0.0

```

- ansible installation method: pip
- ansible-lint installation method: tox dependency

##### OS / ENVIRONMENT
Windows Subsystem for Linux
```
$ uname -a
Linux XXXXXXX 4.4.0-18362-Microsoft #1049-Microsoft Thu Aug 14 12:01:00 PST 2020 x86_64 x86_64 x86_64 GNU/Linux
```
and Jenkins-

##### STEPS TO REPRODUCE
We just need to run `tox -e ansible_lint` as we were doing with 4.3.7 version.

```console 
[testenv:ansible_lint]
whitelist_externals =
    sh
deps =
    ansible-lint
commands =
    ansible-lint {posargs: ./} -vvvv
```

##### Desired Behaviour

Ansible_lint should not report any failure as Ansible module is properly installed with a higher version than needed.

##### Actual Behaviour

```
$ tox -e ansible_lint
ansible_lint installed: ansible-lint==5.0.0,bracex==2.1.1,colorama==0.4.4,commonmark==0.9.1,dataclasses==0.8,enrich==1.2.6,packaging==20.9,Pygments==2.7.4,pyparsing==2.4.7,PyYAML==5.4.1,rich==9.10.0,ruamel.yaml==0.16.12,ruamel.yaml.clib==0.2.2,typing-extensions==3.7.4.3,wcmatch==8.1.1
ansible_lint run-test-pre: PYTHONHASHSEED='3400070394'
ansible_lint run-test: commands[0] | ansible-lint ./ -vvvv
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(colored=True, config_file=None, cwd=PosixPath('/home/{user}/GNAT/ansible-role-doing_stuff'), display_relative_path=True, exclude_paths=['/home/{user}/GNAT/ansible-role-doing_stuff/meta/metadata.yml', '/home/{user}/GNAT/ansible-role-doing_stuff/tests'], format='rich', kinds=[{'requirements': 'requirements.yml'}, {'requirements': '**/meta/requirements.yml'}, {'reno': 'releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{vars,defaults}/*.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'yaml': '**/molecule/*/molecule.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], lintables=['./'], listrules=False, listtags=False, loop_var_prefix=None, mock_modules=[], mock_roles=[], offline=False, parseable=False, parseable_severity=False, progressive=False, quiet=False, rulesdir=[], rulesdirs=['/home/{user}/GNAT/ansible-role-doing_stuff/.tox/ansible_lint/lib/python3.6/site-packages/ansiblelint/rules'], skip_list=['role-name', '204'], tags=[], use_default_rules=False, verbosity=4, version=False, warn_list=['experimental'])
No module named 'ansible'
FATAL: ansible-lint requires a version of Ansible package >= 2.9, but none was found. Please install a compatible version using the same python interpreter. See https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-with-pip
ERROR: InvocationError for command /home/{user}/GNAT/ansible-role-doing_stuff/.tox/ansible_lint/bin/ansible-lint ./ -vvvv (exited with code 4)
____________________________________________
```

**Comment by felixfontein** [comment]
Are both ansible and ansible-lint installed for the same Python version?

**Comment by tadeboro** [comment]
@balmasea Ansible is not automatically installed when installing ansible-lint >= 5.0.0. You need to add community or core extra like this: `pip install ansible-lint[core]`.

**Comment by ssbarnea** [comment]
While this is a local environmental misconfiguration, due to not following the installation guidelines, I am inclined to keep it open as I may need to add some extra sanity checks at instantiation and provide a clear failure message.

The tricky bit is that ansible-lint requires both: ansible-playbook CLI and ansible python module and if the first does not make use of the same ansible python modules you endup with confusing behaviors.

I discovered this because github decided to include ansible by default on their gha images, which means that we detect ansible present, but that may not be present inside the runtime virtualenv.

One can easily guess how this can confuse users.

**Comment by felixfontein** [comment]
@ssbarnea maybe the check could be adapted to also test for the ansible/ansible-playbook binaries if it fails to detect the `ansible` Python module, and if they are found output a different message. That might make it a bit less confusing.

**Comment by tadeboro** [comment]
Yeah, while not strictly ansible-lint's fault, this is definitely a bug because confusing users is not something we should do by default. Detecting and reporting the situation where Ansible CLI is available while python import is not would be a really good start IMHO.

**Comment by balmasea** [comment]
> @balmasea Ansible is not automatically installed when installing ansible-lint >= 5.0.0. You need to add community or core extra like this: `pip install ansible-lint[core]`.

Indeed, that has done it. We just added that package to our tox-deps and it worked. 
```
[testenv:ansible_lint]
whitelist_externals =
    sh
deps =
    ansible-lint[core]
    ansible-lint
commands =
    ansible-lint {posargs: ./}
```

Very much appreciated.

**Comment by balmasea** [comment]
@tadeboro @ssbarnea Indeed, I think confusion comes when it reports that there is no ansible module for the installation. Maybe update doc with `ansible-lint[core]` would be nice, as a suggestion. 

I'll leave it open but our problem has been solved.

**Comment by ssbarnea** [comment]
All the examples including pip mention the extras, so I am closing it. https://ansible-lint.readthedocs.io/en/latest/installing.html
